.github/copilot-instructions.md

```diff
@@ -0,0 +1,229 @@
+# GitHub Copilot Instructions for YASSG
+
+## Repository Overview
+
+YASSG (Yet Another Static Site Generator) is a PHP-based static site generator that combines the power of Twig templating with Symfony Encore for asset management and uses YAML databases for content organization.
+
+## Project Architecture
+
+### Core Components
+
+- **Static Site Generator**: Generates static HTML sites from Twig templates
+- **YAML Database**: Uses YAML files for content and data organization
+- **Asset Pipeline**: Symfony Encore for JavaScript/CSS bundling
+- **CLI Interface**: Console commands for site generation and initialization
+- **Multi-environment**: Supports different build environments (dev, prod)
+
+### Key Directories
+
+- `src/` - Core library code (PSR-4: `Sigwin\YASSG\`)
+- `bin/yassg` - Main CLI entry point
+- `config/` - Symfony configuration files
+- `resources/init/` - Project initialization templates
+  - `basic/` - Minimal project setup
+  - `demo/` - Full-featured demo site
+  - `gitlab/` - GitLab CI/Pages configuration
+- `tests/` - Test suite including functional tests
+- `web/` - Web assets and frontend code
+
+## Development Environment
+
+### Prerequisites
+
+- PHP 8.3+ with ext-zlib
+- Composer for dependency management
+- Docker (for development tools)
+- Make for build automation
+
+### Setup
+
+```bash
+# Install dependencies
+composer install
+
+# Run tests
+make test
+
+# Run code analysis
+make analyze
+
+# Format code
+make dist
+```
+
+### Testing
+
+The project uses PHPUnit with mutation testing via Infection:
+- Unit tests in `tests/unit/`
+- Functional tests in `tests/functional/`
+- Integration tests validate full site generation
+
+### Code Quality
+
+- **PHP-CS-Fixer**: Code formatting (`.php-cs-fixer.dist.php`)
+- **PHPStan**: Static analysis (`phpstan.neon.dist`)
+- **Psalm**: Additional static analysis (`psalm.xml.dist`)
+- **Rector**: Automated refactoring (`rector.php`)
+- **Infection**: Mutation testing (`infection.json.dist`)
+
+## Coding Standards
+
+### PHP Code Style
+
+- PSR-12 coding standard
+- Strict types declaration required: `declare(strict_types=1);`
+- MIT license header in all PHP files
+- Prefer final classes and readonly properties where applicable
+- Use type hints for all parameters and return types
+
+### File Headers
+
+All PHP files should include:
+```php
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sigwin Yassg project.
+ *
+ * (c) sigwin.hr
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+```
+
+### Naming Conventions
+
+- Classes: PascalCase
+- Methods/Properties: camelCase
+- Constants: UPPER_SNAKE_CASE
+- Namespaces follow PSR-4: `Sigwin\YASSG\`
+
+## Key APIs and Components
+
+### Core Classes
+
+- `Generator`: Main site generation logic
+- `Database`: YAML-based data storage interface
```

Comment:
Database is a collection of content, it's not directly done in YAML.

The (filesystem) database can *point* to YAML files, but we need a Decoder instance to actually parse them them.  Database represents a collection of content, YAML Decoder is how a specific piece of content is read.